WordPress/gutenberg
# Unify the PostSchedule component between site and post editors

Related #52632
Follow-up to #56000 
closes #55111

## What?

Now that we're using the EditorProvider in the site editor, we should be able to reuse the exact same component the post editor uses to edit post properties and attributes. In this PR, I'm unifying the "PostSchedule" or "PostPublishDate" component between both.

I'm using the best of both here:

 - I chose to not show the full date but instead went for the site editor format for the toggle.
 - The toggle was missing an aria label from the site editor, which it uses now.

## Testing Instructions

1- Ensure that the changing the page publish date still looks and works well in both post and site editors.
